eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/AggregateBasedStorageEngineTestSuite.java

```diff
@@ -68,22 +72,44 @@ void setUp() throws Exception {
         testSubject = buildStorageEngine();
     }
 
+    @BeforeAll
+    static void beforeAll() {
+        executor = Executors.newVirtualThreadPerTaskExecutor();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        executor.close();
+    }
+
     /**
      * Constructs the {@link AsyncEventStorageEngine} used in this test suite.
      *
      * @return The {@link AsyncEventStorageEngine} used in this test suite.
      */
     protected abstract ESE buildStorageEngine() throws Exception;
 
+    /**
+     * Will translate eventNumber to global sequence position. It differs among the EventStore implementations. For
+     * example: AxonServer starts the global stream from 0, whereas JPA implementations starts from 1.
+     *
+     * @param eventNumber the event number to translate, first is 1
+     * @return the global index position for given event storage engine
+     */
+    protected abstract long globalSequenceOfEvent(long eventNumber);
+
+    protected abstract TrackingToken trackingTokenOnPosition(long eventNumber);
```

Comment:
I think I would rename this method to `trackingTokenAtPosition`. Or, perhaps to `trackingTokenAt(long position)`.
At least, I do think that the parameter name `eventNumber` wasn't intended(?).

Reply:
I'm not sure about the `position` in the name at all (even in the method).
Could you try to help me find better name :) 

Because what we mean by position differ between event store implementations:
the first event -> 0 (AxonServer) or 1 (JPA)
the second event -> 1 (AxonServer) or 2 (JPA)
So I decided that eventNumber is something different, not from the "language" of the specified event store.
Because in the implementation we calculate the position based on which event in order we'd like to receive. 

What do you think about:
`TrackingToken trackingTokenAt(long eventOrder);` 
I may add a documentation what is the meaning of eventOrder and how position of "first" event in order differ?

Reply:
Well, your argument that it means something different per store will stand, regardless of what name you give the parameter.
We could call it `foo` and that would still be true. We could call it `eventOrder` and that's still true. Or we can call it `position`, and that would still be true.

In the end, what it is, is the position of the event in the store.
Or, if you will, the index.

Reply:
Changed